### PR TITLE
Split asset and add column filter

### PR DESCRIPTION
**Pull request by hguenot**

### CODE REVIEW

**Comment by ZAYEC77** [comment]
Thanks for PR.
Looks well for me, but please add some description of your changes and README update

File: src/DataTable.php [hunk 1]
```diff
@@ -1,18 +1,18 @@
 <?php
 /**
  * @copyright Copyright (c) 2015 Serhiy Vinichuk
- * @license MIT
- * @author Serhiy Vinichuk <serhiyvinichuk@gmail.com>
+ * @license   MIT
```
Comment (ZAYEC77) [inline src/DataTable.php]: @hguenot not sure if it's correct format

File: src/DataTableColumn.php [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+ * This file is part of hguenot/base-app project.
+ * =============================================================================
+ * Copyright 2018 - Herve Guenot - https://github.com/hguenot
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ * =============================================================================
+ * 
+ * Date: 25/11/2018
+ * Time: 07:37
```
Comment (ZAYEC77) [inline src/DataTableColumn.php]: @hguenot I guess you could just add `@author` and `@license` doc tags, it will be enough

**Review by ZAYEC77 — COMMENTED** [review]
I think it will be good to create an asset folder in (src) for all asset related classes.
And the same thing for columns.

**Comment by hguenot** [comment]
Hello I've made the change.
I also solved https://github.com/NullRefExcep/yii2-datatables/issues/39 

Now you can make :

```php
    <?php echo \nullref\datatable\DataTable::widget([
        'columns' => [
            'customer.name',
        ],
    ]) ?>
//...
```

This will display field `name` of `customer`.

**Comment by hguenot** [comment]
But before merging I have to made test with `data` attribute...

**Comment by ZAYEC77** [comment]
yeah, sure, thanks

**Comment by hguenot** [comment]
Now it should work !

**Comment by ZAYEC77** [comment]
Thank you, but please check my comments to your code and add readme about this changes

**Comment by hguenot** [comment]
Hello, 

I don't know which change you want me to make.

I've updated README, fix headers. The only thing I've not done is to make asset and column folder because it will break usages.

The main goals are
- using sub-path syntax (`user.name`) instead of writing delegated methods, 
- allow ajax filtering on these subfield
- add column specific filter in header and allow customization of these filters.

**Comment by ZAYEC77** [comment]
@hguenot 
Sorry I missed the update of the readme.

My last question about the parent of `DataTableColumn` class, why did you use `Widget` as the parent, what's reasons for it?

**Comment by hguenot** [comment]
Hi @ZAYEC77 
There is no reason except copy/paste from Yii DataColumn... 
I fixed it 😄

**Comment by hguenot** [comment]
Finally, I revert because I need Widget::getId method to generate id for each filtered input

**Comment by ZAYEC77** [comment]
thank you

**Pull request merged**